docs/src/userguide/tempita.rst

```diff
@@ -0,0 +1,295 @@
+    tmpl2.sub_vars(mapping)        # pass an argument as dict-like mapping.
+
+Tempita supports also shortcut function:
+
+.. code-block:: python
+
+    from Cython.Tempita import sub
+    sub(template_string, **kwargs)
+
+Templating Syntax
+-----------------
+
+Below are the different directive types and expression forms supported.
+
+Simple Expression Substitution
+------------------------------
+
+Surround expression in ``{{ ... }}``, which is evaluated in the template
+context.
+
+Example:
+
+.. code-block:: python
+
+    >>> sub("Hello {{name}}!", name="World")
+    'Hello World!'
+
+Expressions can be any valid Python expression: arithmetic, attribute access,
+indexing, function calls etc.:
+
+.. code-block:: python
+
+    >>> sub("2 * 3 = {{2 * 3}}")
+    '2 * 3 = 6'
+    >>> class User:
+    ...     name = 'Mark'
+    ...
+    >>> sub("User: {{ user.name }}", user=User())
+    'User: Mark'
+    >>> sub("Hex: {{ hex(num) }}", num=10)
+    'Hex: 0xa'
+    >>> sub("Number: {{ func(num) }}", func=abs, num=-10)
+    'Number: 10'
+    >>> sub("List[0] = {{ mylist[0] }}", mylist=[10,20,30])
+    'List[0] = 10'
+    >>> sub("Dist['name'] = {{ mydict['name'] }}", mydict={'name': 'Miss Islington'})
+    "Dist['name'] = Miss Islington"
+    >>> sub("Uppercase {{ string.upper() }}", string="hello")
+    'Uppercase HELLO'
+
+Default Values
+--------------
+
+You can specify a default value for a variable using ``default`` keyword:
+
+.. code-block:: python
+
+    >>> tmpl = Template("""
+    ... {{default name = "Sir Lancelot the Brave"}}
+    ... My name is {{name}}.
+    ... """
+    ... )
+    >>> tmpl.substitute()
+    'My name is Sir Lancelot the Brave.\n'
+    >>> tmpl.substitute(name="Sir Bedevere the Wise")
+    'My name is Sir Bedevere the Wise.\n'
+
+
+Inline Python Code
+------------------
+
+Arbitrary python statements can be inserted inline with ``{{py: ...}}``.
+These do not emit output directly but can modify the template context.
+
+.. code-block:: python
+
+    >>> tmpl = Template("""
+    ... {{py: x = 5}}
+    ... Value of x: {{x}}
+    ... """)
+    >>> tmpl.substitute()
+    'Value of x: 5\n'
+
+Comments
+--------
+
+Comments are introduced with ``{{# ... }}``. They are completely removed
+from output.
+
+.. code-block:: python
+
+    >>> sub("Hello {{# This is a comment }}World.")
+    'Hello World.'
+
+Custom delimiters
+-----------------
+
+If you want literal text that looks like a directive (e.g. `{{...}}`), you
+can use custom delimiters:
+
+.. code-block:: python
+
+    >>> sub("Show literal braces: {{((name))}}", delimiters=['((', '))'], name="x")
+    'Show literal braces: {{x}}'
+
+.. note:: Parameter `delimiters` can be also passed to ``Template`` class as an argument:
+
+   ``Template("...", delimiters=['((', '))'])``
+
+Filters
+-------
+
+You can pass callable which will be automatically called with expression as parameter:
+
+.. code-block:: python
+
+    {{ expression | filtername }}
```

Comment:
This does imply

> Expressions can be any valid Python expression

may not be quite true.

(Don't think this needs changes to the PR though)